### PR TITLE
Fix Makefile install instructions and improve CONTRIBUTING.md

### LINKED ISSUE #1820: Improve CONTRIBUTING.md and update Makefile help message
Hi 👋🏼 

The CONTRIBUTING.md file doesn't mention the requirements (golangci-lint and richgo), we should add those for a smoother experience.

And while we are on that, when I tried to run `make test` the first time, this is what it looked like:

```shell
Makefile:5: "could not find golangci-lint in /opt/homebrew/opt/llvm/bin:/opt/homebrew/opt/curl/bin:/opt/homebrew/opt/libiconv/bin:/Users/maxrn/.pyenv/shims:/Users/maxrn/.nvm/versions/node/v16.13.0/bin:/Library/Frameworks/Python.framework/Versions/2.7/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/TeX/texbin:/usr/local/go/bin:/usr/local/MacGPG2/bin:/usr/local/share/dotnet:/opt/X11/bin:~/.dotnet/tools:/Library/Apple/usr/bin:/Library/Frameworks/Mono.framework/Versions/Current/Commands:/Users/maxrn/.sdkman/candidates/java/current/bin:/opt/homebrew/opt/llvm/bin:/Users/maxrn/.cabal/bin:/Users/maxrn/.ghcup/bin:/opt/homebrew/opt/curl/bin:/opt/homebrew/opt/libiconv/bin:/Users/maxrn/.nvm/versions/node/v16.13.0/bin:/Library/Frameworks/Python.framework/Versions/2.7/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/Users/maxrn/.cargo/bin:/opt/homebrew/opt/fzf/bin:/Users/maxrn/go/bin:/Users/maxrn/src/sonar-scanner-4.7.0.2747-macosx/bin:/Users/maxrn/.dotfiles/bin:/Users/maxrn/go/bin:/Users/maxrn/src/sonar-scanner-4.7.0.2747-macosx/bin:/Users/maxrn/.dotfiles/bin, run: curl -sfL https://install.goreleaser.com/github.com/golangci/golangci-lint.sh | sh"
Makefile:9: "could not find richgo in /opt/homebrew/opt/llvm/bin:/opt/homebrew/opt/curl/bin:/opt/homebrew/opt/libiconv/bin:/Users/maxrn/.pyenv/shims:/Users/maxrn/.nvm/versions/node/v16.13.0/bin:/Library/Frameworks/Python.framework/Versions/2.7/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/TeX/texbin:/usr/local/go/bin:/usr/local/MacGPG2/bin:/usr/local/share/dotnet:/opt/X11/bin:~/.dotnet/tools:/Library/Apple/usr/bin:/Library/Frameworks/Mono.framework/Versions/Current/Commands:/Users/maxrn/.sdkman/candidates/java/current/bin:/opt/homebrew/opt/llvm/bin:/Users/maxrn/.cabal/bin:/Users/maxrn/.ghcup/bin:/opt/homebrew/opt/curl/bin:/opt/homebrew/opt/libiconv/bin:/Users/maxrn/.nvm/versions/node/v16.13.0/bin:/Library/Frameworks/Python.framework/Versions/2.7/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/Users/maxrn/.cargo/bin:/opt/homebrew/opt/fzf/bin:/Users/maxrn/go/bin:/Users/maxrn/src/sonar-scanner-4.7.0.2747-macosx/bin:/Users/maxrn/.dotfiles/bin:/Users/maxrn/go/bin:/Users/maxrn/src/sonar-scanner-4.7.0.2747-macosx/bin:/Users/maxrn/.dotfiles/bin, run: go install github.com/kyoh86/richgo@latest"
******************** downloading dependencies ********************
go get -v ./...
******************** running tests ********************
richgo test -v ./...
make: richgo: No such file or directory
make: *** [test] Error 1
```

We are printing the entire PATH variable which is pointless IMO. It makes it hard to parse the actual error and the help message ((warning "could not find golangci-lint in $(PATH), run: curl -sfL https://install.goreleaser.com/github.com/golangci/golangci-lint.sh | sh")) is also easily glossed over. I would propose not printing the value of $PATH but just "$PATH". This makes clear the error and also helps the user see the help message more easily.

On that note, the link for installing golangci-lint is outdated. According to the [documentation](https://golangci-lint.run/usage/install/#local-installation) the recommended way for Mac is using brew and the curl command is now `
curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.50.0
`

### PR DESCRIPTION
**Pull request by FollowTheProcess**
Closes #1820 

This PR addresses the concerns raised in the linked issue above re. Makefile printing entire `$PATH` variable and incorrect instructions for installing golangci-lint.

I've also added a line in `CONTRIBUTING.md` linking to the two dev tools required.

On running linting there was a few warnings about deprecated and/or abandoned linters so I've fixed them up:

```shell
WARN [runner] The linter 'varcheck' is deprecated (since v1.49.0) due to: The owner seems to have abandoned the linter.  Replaced by unused. 
WARN [runner] The linter 'interfacer' is deprecated (since v1.38.0) due to: The repository of the linter has been archived by the owner.  
WARN [runner] The linter 'maligned' is deprecated (since v1.38.0) due to: The repository of the linter has been archived by the owner.  Replaced by govet 'fieldalignment'. 
WARN [runner] The linter 'deadcode' is deprecated (since v1.49.0) due to: The owner seems to have abandoned the linter.  Replaced by unused. 
WARN [runner] The linter 'structcheck' is deprecated (since v1.49.0) due to: The owner seems to have abandoned the linter.  Replaced by unused. 
WARN [runner] The linter 'golint' is deprecated (since v1.41.0) due to: The repository of the linter has been archived by the owner.  Replaced by revive. 
```

I've made the suggested replacements (most were replaced by `unused`)

### CODE REVIEW

**Comment by umarcor** [comment]
@FollowTheProcess can you please rebase on top of #1785 ?

**Review by makeavish — APPROVED** [review]

File: .golangci.yml [hunk 1]
```diff
@@ -33,30 +33,31 @@ linters:
...
     #- goprintffuncname
     #- gosec
     #- gosimple
     - govet
     - ineffassign
-    - interfacer
+    # - interfacer
     #- lll
-    - maligned
+    # - maligned
```
Comment (marckhouzam) [inline .golangci.yml:45]: I don't think we are running the replacement linter for this one.
According to the `golangci-lint` warning: "Replaced by govet 'fieldalignment'"

If I make the following modification without the PR I get an linter error, but not after the PR.

```
diff --git a/command.go b/command.go
index 9d5e9cf..943bb17 100644
--- a/command.go
+++ b/command.go
@@ -175,9 +175,6 @@ type Command struct {
 	// helpCommandGroupID is the group id for the helpCommand
 	helpCommandGroupID string

-	// completionCommandGroupID is the group id for the completion command
-	completionCommandGroupID string
-
 	// versionTemplate is the version template defined by user.
 	versionTemplate string

@@ -194,6 +191,8 @@ type Command struct {
 	// CompletionOptions is a set of options to control the handling of shell completion
 	CompletionOptions CompletionOptions

+	// completionCommandGroupID is the group id for the completion command
+	completionCommandGroupID string
 	// commandsAreSorted defines, if command slice are sorted or not.
 	commandsAreSorted bool
 	// commandCalledAs is the name or alias value used to call this command.
```

Comment (FollowTheProcess) [reply .golangci.yml:45]: Yeah I think we'd need to configure govet to do that: https://golangci-lint.run/usage/linters/#govet. I tend to just use:

```yaml
  govet:
    enable-all: true
```

But if we've not run that before it might trigger a load of stuff

**Comment by FollowTheProcess** [comment]
> @FollowTheProcess can you please rebase on top of https://github.com/spf13/cobra/pull/1785 ?

Ahh sorry I didn't see there was already one addressing the linter config. If it's easier and keeps this PR cleaner I can remove the linter changes from this one?

**Comment by marckhouzam** [comment]
> Ahh sorry I didn't see there was already one addressing the linter config. If it's easier and keeps this PR cleaner I can remove the linter changes from this one?

Yeah, if we keep the changes separate, we will be able to merge more easily.  Thanks 👍

File: Makefile [hunk 2]
```diff
@@ -1,12 +1,13 @@
 BIN="./bin"
 SRC=$(shell find . -name "*.go")
+GOPATH=$(shell go env GOPATH)
 
 ifeq (, $(shell which golangci-lint))
-$(warning "could not find golangci-lint in $(PATH), run: curl -sfL https://install.goreleaser.com/github.com/golangci/golangci-lint.sh | sh")
+$(warning "could not find golangci-lint in PATH, run: curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(GOPATH)/bin v1.50.0")
```
Comment (FollowTheProcess) [inline Makefile:6]: My only concern here is that the command now specifies a version of the linter to install (which would have to be kept up to date). Maybe it's better to simply point to the website which has good install docs: https://golangci-lint.run/usage/install/

Comment (marckhouzam) [reply Makefile:6]: > My only concern here is that the command now specifies a version of the linter to install (which would have to be kept up to date). Maybe it's better to simply point to the website which has good install docs: https://golangci-lint.run/usage/install/

Agreed.

Comment (umarcor) [reply Makefile:6]: An alternative is to use go install, as done in CI. However, that won't add it to the PATH.

**Comment by FollowTheProcess** [comment]
Please don't stale 😂 do I need to do anything to help you guys be able to merge this?

**Comment by umarcor** [comment]
@FollowTheProcess, see https://github.com/spf13/cobra/issues/1858. You can ignore those comments. Furthermore, you might want to propose a PR to remove the stale-bot. Removing https://github.com/spf13/cobra/blob/main/.github/workflows/stale.yml should suffice.
